Source/Particles/Resampling/VelocityCoincidenceThinning.cpp

```diff
@@ -0,0 +1,284 @@
+/* Copyright 2024 The WarpX Community
+ *
+ * This file is part of WarpX.
+ *
+ * Authors: Roelof Groenewald (TAE Technologies)
+ *
+ * License: BSD-3-Clause-LBNL
+ */
+
+#include "VelocityCoincidenceThinning.H"
+
+#include "Particles/Algorithms/KineticEnergy.H"
+#include "Utils/Parser/ParserUtils.H"
+#include "Utils/ParticleUtils.H"
+
+
+VelocityCoincidenceThinning::VelocityCoincidenceThinning (const std::string& species_name)
+{
+    using namespace amrex::literals;
+
+    const amrex::ParmParse pp_species_name(species_name);
+
+    utils::parser::queryWithParser(
+        pp_species_name, "resampling_min_ppc", m_min_ppc
+    );
+    WARPX_ALWAYS_ASSERT_WITH_MESSAGE(
+        m_min_ppc >= 1,
+        "Resampling min_ppc should be greater than or equal to 1"
+    );
+
+    utils::parser::queryWithParser(
```

Comment:
These parameters will be very much application dependent. Perhaps make these required inputs to avoid surprises for users. This also forces the user to have some consideration of what is happening. The documentation can give guidelines for what values to use.

Reply:
Sounds good! I agree, default values here could be problematic. I left the default `min_ppc=1` since that is also used by `LevelingThinning`.